src/encoded/types/file.py

```diff
@@ -492,24 +495,32 @@ def download(context, request):
     proxy = asbool(request.params.get('proxy')) or 'Origin' in request.headers \
                                                 or 'Range' in request.headers
 
-    use_download_proxy = request.client_addr not in request.registry['aws_ipset']
+    use_download_proxy = request.client_addr and request.client_addr not in request.registry['aws_ipset']
 
     external = context.propsheets.get('external', {})
-    if external.get('service') == 's3':
-        conn = boto3.client('s3')
-        location = conn.generate_presigned_url(
-            ClientMethod='get_object',
-            Params={
-                'Bucket': external['bucket'],
-                'Key': external['key'],
-                'ResponseContentDisposition': 'attachment; filename=' + filename
-            },
-            ExpiresIn=36*60*60
-        )
-    else:
-        raise HTTPNotFound(
-            detail='External service {} not expected'.format(external.get('service'))
-        )
+    if not external.get('service') == 's3':
```

Comment:
@caseylitton But now you are trying to use boto3 even if the service isn't s3?

Reply:
I think `if not external.get('service') == 's3'`, this download method will either return a local file or raise `HTTPNotFound` or other exceptions. So when it tries to use boto3, it should have confimed `external.get('service') == 's3'`. Do I understand the code correctly? And should we make it more explicit/clear?

Reply:
Ah, was missing the HTTPNotFound before. Thanks, makes sense.

Reply:
No problem. Thank you for taking a look here.